openslides/core/static/js/core/base.js

```diff
@@ -223,22 +223,40 @@ angular.module('OpenSlidesApp.core', [
             } catch(err) {
                 console.error(json);
             }
-            _.forEach(dataList, function(data) {
-                console.log("Received object: " + data.collection + ", " + data.id);
-                var instance = DS.get(data.collection, data.id);
-                if (data.action == 'changed') {
+
+            var dataList_byCollection = _.groupBy(dataList, 'collection');
+            var counter = 0;
+            _.forEach(dataList_byCollection, function(list, key) {
+                var changedElements = [];
+                var deletedElements = [];
+                var collectionString = key;
+                _.forEach(list, function(data) {
+                    // uncomment this line for debugging to log all autoupdates:
+                    // console.log("Received object: " + data.collection + ", " + data.id);
+
+                    // remove (=eject) object from local DS store
+                    var instance = DS.get(data.collection, data.id);
                     if (instance) {
-                        // The instance is in the local db
                         dsEject(data.collection, instance);
                     }
-                    DS.inject(data.collection, data.data);
-                } else if (data.action == 'deleted') {
-                    if (instance) {
-                        // The instance is in the local db
-                        dsEject(data.collection, instance);
+                    // object changed
+                    if (data.action == 'changed') {
+                        changedElements.push(data.data);
                     }
-                    DS.eject(data.collection, data.id);
+                    // object deleted
```

Comment:
Please use `else if` clause for the delete case (also with `===`). If the message is something else the changed or delete we should print a `console.error`.